File: Dopamine.ControlsModule/ViewModels/NowPlayingPlaybackControlsViewModel.vb
Imports Microsoft.Practices.Prism.Mvvm
Imports Dopamine.Core.Utils
Imports Dopamine.Common.Services.Playback
Imports Dopamine.Common.Presentation.ViewModels

Namespace ViewModels
    Public Class NowPlayingPlaybackControlsViewModel
        Inherits BindableBase

#Region "Variables"
        Private mPlaybackInfoViewModel As PlaybackInfoViewModel
        Private mPlaybackService As IPlaybackService
#End Region

#Region "ReadOnly Properties"
        Public ReadOnly Property IsShowcaseAvailable() As Boolean
            Get
                Return Not mPlaybackService.IsStopped
            End Get
        End Property
#End Region

#Region "Properties"
        Public Property PlaybackInfoViewModel() As PlaybackInfoViewModel
            Get
                Return mPlaybackInfoViewModel
            End Get
            Set(ByVal value As PlaybackInfoViewModel)
                SetProperty(Of PlaybackInfoViewModel)(Me.mPlaybackInfoViewModel, value)
            End Set
        End Property
#End Region

#Region "Construction"
        Public Sub New(iPlaybackService As IPlaybackService)
            Me.mPlaybackService = iPlaybackService

            AddHandler Me.mPlaybackService.PlaybackProgressChanged, Sub() Me.UpdateTime()
            AddHandler Me.mPlaybackService.PlaybackStopped, Sub()
                                                                Me.Reset()
                                                                OnPropertyChanged(Function() Me.IsShowcaseAvailable)
                                                            End Sub
            AddHandler mPlaybackService.PlaybackFailed, Sub() OnPropertyChanged(Function() Me.IsShowcaseAvailable)
            AddHandler mPlaybackService.PlaybackPaused, Sub() OnPropertyChanged(Function() Me.IsShowcaseAvailable)
            AddHandler mPlaybackService.PlaybackResumed, Sub() OnPropertyChanged(Function() Me.IsShowcaseAvailable)
            AddHandler mPlaybackService.PlaybackSuccess, Sub() OnPropertyChanged(Function() Me.IsShowcaseAvailable)

            Me.Reset()
        End Sub
#End Region

#Region "Private"
        Private Sub UpdateTime()
            Me.PlaybackInfoViewModel.CurrentTime = FormatUtils.FormatTime(Me.mPlaybackService.GetCurrentTime)
            Me.PlaybackInfoViewModel.TotalTime = " / " & FormatUtils.FormatTime(Me.mPlaybackService.GetTotalTime)
        End Sub

        Private Sub Reset()
            Me.PlaybackInfoViewModel = New PlaybackInfoViewModel With {.Title = "", .Artist = "", .Album = "", .Year = "", .CurrentTime = "", .TotalTime = ""}
        End Sub
#End Region
    End Class
End Namespace

File: Dopamine.ControlsModule/ViewModels/NowPlayingPlaybackControlsViewModel.cs
using Dopamine.Common.Enums;
using Dopamine.Common.Services.Playback;
using Dopamine.Core.Settings;

namespace Dopamine.ControlsModule.ViewModels
{
    public class NowPlayingPlaybackControlsViewModel : PlaybackControlsViewModelBase
    {
        #region Variables
        private bool isShowCaseSelected;
        private bool isPlaylistSelected;
        private bool isLyricsSelected;
        private bool isArtistInformationSelected;
        #endregion

        #region Properties
        public bool IsShowCaseSelected
        {
            get { return this.isShowCaseSelected; }
            set { SetProperty<bool>(ref this.isShowCaseSelected, value); }
        }

        public bool IsPlaylistSelected
        {
            get { return this.isPlaylistSelected; }
            set { SetProperty<bool>(ref this.isPlaylistSelected, value); }
        }

        public bool IsLyricsSelected
        {
            get { return this.isLyricsSelected; }
            set { SetProperty<bool>(ref this.isLyricsSelected, value); }
        }

        public bool IsArtistInformationSelected
        {
            get { return this.isArtistInformationSelected; }
            set { SetProperty<bool>(ref this.isArtistInformationSelected, value); }
        }

        public bool HasPlaybackQueue
        {
            get { return this.playbackService.Queue.Count > 0; }
        }

        #endregion

        #region Construction
        public NowPlayingPlaybackControlsViewModel(IPlaybackService playbackService) : base(playbackService)
        {
            this.playbackService.PlaybackSuccess += (_) => OnPropertyChanged(() => this.HasPlaybackQueue);

            this.playbackService.PlaybackStopped += (_, __) =>
            {
                this.Reset();
            };

            this.SelectMenuItem();
        }
        #endregion

        #region Private
        private void SelectMenuItem()
        {
            if (XmlSettingsClient.Instance.Get<bool>("Startup", "ShowLastSelectedPage"))
            {
                SelectedNowPlayingPage screen = (SelectedNowPlayingPage)XmlSettingsClient.Instance.Get<int>("FullPlayer", "SelectedNowPlayingPage");

                switch (screen)
                {
                    case SelectedNowPlayingPage.ShowCase:
                        this.IsShowCaseSelected = true;
                        break;
                    case SelectedNowPlayingPage.Playlist:
                        this.IsPlaylistSelected = true;
                        break;
                    case SelectedNowPlayingPage.Lyrics:
                        this.IsLyricsSelected = true;
                        break;
                    case SelectedNowPlayingPage.ArtistInformation:
                        this.IsArtistInformationSelected = true;
                        break;
                }
            }
            else
            {
                this.IsPlaylistSelected = true;
            }
        }
        #endregion
    }
}
